### PR TITLE
Fix DBConfig used in builder

### PR DESCRIPTION
**Pull request by ZexiLiu**

### CODE REVIEW

File: include/libjungle/jungle_builder.h [hunk 1]
```diff
@@ -44,6 +44,11 @@ class Builder {
          */
         std::string path;
 
+        /**
+         * DBConfig for the DB instance to be created.
+         */
+        const DBConfig* dbConfig;
```
Comment (greensky00) [inline include/libjungle/jungle_builder.h:50]: It should be `DBConfig dbConfig`, not pointer. The caller (of `init`) life span may not be longer than builder's life span.

Comment (ZexiLiu) [reply include/libjungle/jungle_builder.h:50]: In the init() function, https://github.com/eBay/Jungle/pull/177/files#diff-e81fa7970a0833b1cb0b4bdbfadfefa84eaee52400bbf5d0b49d47bbe5474e63R121-R124

We already copy the passed in db_config to the dbConfig member of `Builder`, the pointer in this `BuildParams` will have the same life span as `Builder`.

Comment (greensky00) [reply include/libjungle/jungle_builder.h:50]: Ok, now I get it.

**Review by greensky00 — APPROVED** [review]

**Review by greensky00 — CHANGES_REQUESTED** [review]
Wait, `dbConfig` should be initialized to null.

**Comment by ZexiLiu** [comment]
> Wait, `dbConfig` should be initialized to null.

I change it from pointer to regular object so that BuilderParams will have a default db config.

**Review by greensky00 — APPROVED** [review]

**Pull request merged**